guvnor-webapp/src/main/java/org/guvnor/client/perspectives/M2RepoPerspective.java

```diff
@@ -20,13 +20,13 @@
 import javax.enterprise.event.Event;
```

Comment:
@Hasys I had commented:

"New instances of the View will be created "on demand". Therefore the View instance @Inject'ed into the Presenter will be a different instance to that you create here. You should really add a method to the Presenter to add a HideHandler to the View and only instantiate and destroy the Presenter here."

But you still seem to be creating the View to add the ```HideHandler```